service/src/main/kotlin/fi/espoo/evaka/incomestatement/IncomeStatementQueries.kt

```diff
@@ -447,3 +447,10 @@ fun Database.Read.isOwnIncomeStatement(user: AuthenticatedUser.Citizen, id: Inco
         .bind("personId", user.id)
         .mapTo<Int>()
         .any()
+
+fun Database.Read.incomeStatementExistsForStartDate(user: AuthenticatedUser.Citizen, startDate: LocalDate): Boolean =
+    createQuery("SELECT EXISTS(SELECT 1 FROM income_statement WHERE person_id = :personId AND start_date = :startDate)")
```

Comment:
Fixed, although I think `SELECT EXISTS(...)` would be the better default to use. I guess the database could optimize it better, because with `.mapTo<Int>().any()` it doesn't know that we're only interested in whether at least one result exists.

Reply:
Very true. This one was mainly nitpicking.